phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -2957,149 +2958,180 @@ public boolean isViewReferenced() {
                 splits = SchemaUtil.processSplits(splits, pkColumns, saltBucketNum, connection.getQueryServices().getProps().getBoolean(
                         QueryServices.FORCE_ROW_KEY_ORDER_ATTRIB, QueryServicesOptions.DEFAULT_FORCE_ROW_KEY_ORDER));
             }
-            MetaDataMutationResult result = connection.getQueryServices().createTable(
-                    tableMetaData,
-                    viewType == ViewType.MAPPED || allocateIndexId ? physicalNames.get(0).getBytes() : null,
-                    tableType, tableProps, familyPropList, splits, isNamespaceMapped, allocateIndexId,
-                    UpgradeUtil.isNoUpgradeSet(connection.getClientInfo()), parent);
+
+			// Modularized this code for unit testing
+            MetaDataMutationResult result = getCreateTableMutationResult(tableMetaData, viewType,
+                    allocateIndexId, physicalNames, tableType, tableProps, familyPropList, splits,
+                    isNamespaceMapped, parent);
             MutationCode code = result.getMutationCode();
-            switch(code) {
+            if (code != MutationCode.TABLE_NOT_FOUND) {
+                boolean mutationResult =  handleCreateTableMutationCode(result, code, statement,
```

Comment:
mutationResult doesn't look like the correct name from readability perspective. Something like tableAreadyExists might be better. From a glance at the function it looks like it returns true if thats the case.